### PR TITLE
Fix storage location of Docker images and containers for Mac or linux

### PR DESCRIPTION
**Pull request by DamMicSzm**
Problem
Information about used memory space when build image of docker on osx not working

Change overview

For linux  docker volume dir is /var/lib/docker/
For Mac docker volume dir is ~/Library/Containers/com.docker.docker/Data/vms/0/
A conditional statement has been added to check the system type.

Testing
Run ./integrations/docker/images/chip-build-tizen/build.sh
or other images

### CODE REVIEW

File: integrations/docker/build.sh [hunk 1]
```diff
@@ -59,7 +59,12 @@ set -ex
 
 [[ -n $VERSION ]] || die "version cannot be empty"
 
-mb_space_before=$(df -m /var/lib/docker/ | awk 'FNR==2{print $3}')
+#Fix storage location of Docker images and containers for osx or linux
```
Comment (andy31415) [inline integrations/docker/build.sh:62]: Could we update the comment? I am unsure how `Fix the storage location...` helps me read the code. I read the code as gdarwin and non-darwin have different commands to compute docker space left`

Lets drop the entire comment if nothing can be added to enhance code readability. The most I could think of would be that these rely on default paths for docker (can one set non-default) but that also seems obvious reading the code.

File: integrations/docker/build.sh [hunk 2]
```diff
@@ -91,9 +96,16 @@ docker image prune --force
     }
 }
 
+#Fix storage location of Docker images and containers for osx or linux
```
Comment (andy31415) [inline integrations/docker/build.sh:99]: drop this comment

File: integrations/docker/build.sh [hunk 3]
```diff
@@ -59,7 +59,12 @@ set -ex
 
 [[ -n $VERSION ]] || die "version cannot be empty"
 
-mb_space_before=$(df -m /var/lib/docker/ | awk 'FNR==2{print $3}')
+#Fix storage location of Docker images and containers for osx or linux
+if [[ $OSTYPE == 'darwin'* ]]; then
+    mb_space_before=$(df -m ~/Library/Containers/com.docker.docker/Data/vms/0/ | awk 'FNR==2{print $3}')
+else
+    mb_space_before=$(df -m /var/lib/docker/ | awk 'FNR==2{print $3}')
```
Comment (bzbarsky-apple) [inline integrations/docker/build.sh:66]: Shouldn't we just set a variable up front based on `$OSTYPE` that points to the right location, and then we don't have to have these branches and the copy/pasted paths in multiple places....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: integrations/docker/build.sh (hunk 1 above)
Comment (DamMicSzm) [reply integrations/docker/build.sh:62]: Thank you for your response. Of course you are right, the comment is redundant. Thanks

File: integrations/docker/build.sh (hunk 2 above)
Comment (DamMicSzm) [reply integrations/docker/build.sh:99]: The comment will be deleted. Thanks for the review :)

File: integrations/docker/build.sh (hunk 3 above)
Comment (DamMicSzm) [reply integrations/docker/build.sh]: A very valid point. I will correct it. Thank you for your attention and review. :)

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@DamMicSzm Thank you for the fix!

**Pull request merged**